packages/aws-cdk/test/api/hotswap/hotswap-deployments.test.ts

```diff
@@ -317,3 +317,42 @@ test('can correctly reference AWS::URLSuffix in hotswappable changes', async ()
   expect(hotswapMockSdkProvider.mockSdkProvider.sdk.removeCustomUserAgent)
     .toHaveBeenCalledWith('cdk-hotswap/success-lambda-function');
 });
+
+test('attempting to hotswap a newly created resource with the same logical ID that a resource which is currently deployed to CloudFormation has results in a full deployment', async () => {
```

Comment:
This description is weird, and I don't think is very grammatical 😛. How about

```suggestion
test('changing the type of a deployed resource always results in a full deployment', async () => {
```